### ISSUE TITLE
Bug: `clippy::from-iter-instead-of-collect` for `FxHashSet`

### ISSUE THREAD

**Issue by nahuakang** [rust-lang/rust-clippy#9315]
### Description

# Potential Bug

While trying to create a new lint, I wrote [this line](https://github.com/rust-lang/rust-clippy/blob/812db454b862b3604d6dca812cdc3c7bc740f635/clippy_lints/src/methods/collapsible_str_replace.rs#L197):

```rust
let to_arg_repr_set = FxHashSet::from_iter(to_arg_reprs.iter().cloned());
```

The feedback I received was:

```sh
error: usage of `FromIterator::from_iter`
   --> src/methods/collapsible_str_replace.rs:194:27
    |
194 |     let to_arg_repr_set = FxHashSet::from_iter(to_arg_reprs.iter().cloned());
    |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `.collect()` instead of `::from_iter()`: `to_arg_reprs.iter().cloned().collect::<FxHashSet<_, _>>()`
    |
    = note: `-D clippy::from-iter-instead-of-collect` implied by `-D clippy::pedantic`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#from_iter_instead_of_collect
```

Unfortunately that's not how `FxHashSet` works: https://doc.rust-lang.org/beta/nightly-rustc/rustc_data_structures/fx/type.FxHashSet.html. So `to_arg_reprs.iter().cloned().collect::<FxHashSet<_, _>>()` does not really apply here and will get another compiler error (or rust-analyzer warning).

What I should have written was:

```rust
let to_arg_repr_set = to_arg_reprs.iter().cloned().collect::<FxHashSet<_>>();
```

The lint suggestion that I should have received would be:

```sh
error: usage of `FromIterator::from_iter`
   --> src/methods/collapsible_str_replace.rs:194:27
    |
194 |     let to_arg_repr_set = FxHashSet::from_iter(to_arg_reprs.iter().cloned());
    |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `.collect()` instead of `::from_iter()`: `to_arg_reprs.iter().cloned().collect::<FxHashSet<_>>()`
    |
    = note: `-D clippy::from-iter-instead-of-collect` implied by `-D clippy::pedantic`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#from_iter_instead_of_collect
```

### Version

```text
rustc 1.61.0 (fe5b13d68 2022-05-18)
binary: rustc
commit-hash: fe5b13d681f25ee6474be29d748c65adcd91f69e
commit-date: 2022-05-18
host: aarch64-apple-darwin
release: 1.61.0
LLVM version: 14.0.0
```
